src/vmm/src/device_manager/persist.rs

```diff
@@ -766,6 +824,37 @@ mod tests {
                 .serialize(&mut buf.as_mut_slice(), &version_map, 3)
                 .unwrap();
 
+            // Add an entropy device.
+            let entropy_config = EntropyDeviceConfig {};
+            insert_entropy_device(&mut vmm, &mut cmdline, &mut event_manager, entropy_config);
+
+            version_map
+                .new_version()
+                .set_type_version(DeviceStates::type_id(), 4);
+
+            // Entropy device not supported in version < 4
+            assert_eq!(
+                vmm.mmio_device_manager
+                    .save()
+                    .serialize(&mut buf.as_mut_slice(), &version_map, 3),
+                Err(VersionizeError::Semantic(
+                    "Target version does not support persisting the virtio-rng device.".to_string()
+                ))
+            );
+
+            version_map
+                .new_version()
+                .set_type_version(DeviceStates::type_id(), 4)
+                .set_type_version(NetConfigSpaceState::type_id(), 2);
+
+            vmm.mmio_device_manager
+                .save()
+                .serialize(&mut buf.as_mut_slice(), &version_map, 4)
+                .unwrap();
+            let device_states: DeviceStates =
+                DeviceStates::deserialize(&mut buf.as_slice(), &version_map, 4).unwrap();
+            assert!(device_states.entropy_device.is_some());
```

Comment:
Nit it is usually better to match the specific value e.g. `assert_eq!(..., Some(...))`.

Reply:
Hmmm, interesting. It sounds counter-intuitive to me, since I don't really care about what's inside `Some`. Could you elaborate?

Reply:
Take an example:
```
fn outer_fn() -> Option<i32> {
    Some(inner_fn())
}
fn inner_fn() -> i32 {
    4
}
```
If we assume `inner_fn()` is pre-existing and that we are adding `outer_fn()` we may test this by `assert!(outer_fn().is_some())`  but here we are missing something. `outer_fn()` could definitely mutate the the value from `inner_fn()` incorrectly and still return `Some` so it would be better to test `assert_eq!(outer_fn(),Some(4))` as we test the full behavior of `outer_fn()`.

Testing `assert!(outer_fn().is_some())` is like testing `assert!(inner_fn() > 0)` (in fact if `inner_fn()` returned `NonZeroI32` these would compile to the exact same) when it would be better to test the exact value.

Reply:
I understand that we should check the value contained in `Some` if we cared about it.
Since here we: (1) rely on `Versionize` to do the serialization/deserialization, (2) it's not quite straight-forward to get what's inside `Some` and we actually check for the expected VM configuration later on, I think it's not needed in this particular assertion.

Thoughts?

Reply:
It is always better to check the value, but if its not worth the effort in this case I think that's reasonable.